parquet-cli/src/main/java/org/apache/parquet/cli/commands/CheckParquet251Command.java

```diff
@@ -340,7 +340,7 @@ private void validateStatsForPage(DataPage page, DictionaryPage dict,
         column.consume();
       }
 
-      if (numNulls != stats.getNumNulls()) {
+      if (stats.isNumNullsSet() && numNulls != stats.getNumNulls()) {
```

Comment:
this looks like a bug in upstream logic - if there are no nulls `stats.getNumNulls()` returns -1

Reply:
We hit this since we added a wider range for shouldIgnoreStatistics as we removed some of the boundary checksin our fork